packages/components/src/components/post-popover/post-popover.tsx

```diff
@@ -0,0 +1,169 @@
+      trigger.addEventListener('keypress', this.localEnterTogglePopover);
+      trigger.addEventListener('touch', this.localTouchTogglePopover, { once: true });
+      trigger.setAttribute('aria-expanded', 'false');
+    });
+  }
+
+  disconnectedCallback() {
+    this.triggers.forEach(trigger => {
+      trigger.removeEventListener('mouseup', this.localTogglePopover);
+      trigger.removeEventListener('keypress', this.localEnterTogglePopover);
+      trigger.removeEventListener('touch', this.localTouchTogglePopover);
+      trigger.removeAttribute('aria-expanded');
+    });
+  }
+
+  /**
+   * Programmatically display the popover
+   * @param target An element with [data-popover-target="id"] where the popover should be shown
+   */
+  @Method()
+  async show(target: HTMLElement) {
+    this.currentTarget = target;
+    this.popoverRef.show(target);
+    target.setAttribute('aria-expanded', 'true');
+  }
+
+  /**
+   * Programmatically hide this popover
+   */
+  @Method()
+  async hide() {
+    this.popoverRef.hide();
+    this.triggers.forEach(trigger => trigger.setAttribute('aria-expanded', 'false'));
+  }
+
+  /**
+   * Toggle popover display
+   * @param target An element with [data-popover-target="id"] where the popover should be anchored to
+   * @param force Pass true to always show or false to always hide
+   */
+  @Method()
+  async toggle(target: HTMLElement, force?: boolean) {
+    this.currentTarget = target;
+    this.popoverRef.toggle(target, force);
+  }
+
+  private get triggers() {
+    return document.querySelectorAll(`[data-popover-target="${this.host.id}"]`);
+  }
+
+  /**
+   * One time event handler for click events
+   * A permanent event listener would prevent a toggle button from working properly:
+   * A click opens the popover, a second click first closes it (due to light dismiss), then directly
+   * opens it again because of the click listener on the button. Registering a new
+   * one time listener after a small timeout solves this issue.
+   * @param e toggle event from post-popovercontainer
+   */
+  private onToggle(e: CustomEvent<boolean>) {
+    if (this.currentTarget) {
+      this.currentTarget.setAttribute('aria-expanded', `${e.detail}`);
+    }
+    if (!e.detail) {
+      window.requestAnimationFrame(() => {
+        this.triggers.forEach(trigger => {
+          trigger.addEventListener('mouseup', this.localTogglePopover, { once: true });
+          trigger.addEventListener('touch', this.localTouchTogglePopover, { once: true });
+        });
+      });
+
+      // Handle missing re-focusing logic after close. Can be removed as soon as popovertarget works correctly
+      if (this.currentTarget) {
+        this.currentTarget.focus();
+        this.currentTarget = null;
+      }
+    }
+  }
+
+  // Fix for firefox to prevent the following lines from triggering
+  // https://github.com/oddbird/popover-polyfill/blob/main/src/popover.ts#L338
```

Comment:
Can you detail why we need it, please? When I try the demos of oddbird/popover-polyfill, I have no problem, so I guess this has to do with cross-shadow-boundary?

Reply:
I have no idea what the issue is. Event bubbling through shadow dom is a bit tricky, that may be the cause. The standard API for triggering will also not support cross shadow root triggers (which is a shame: https://codepen.io/tuelsch/pen/LYMvmva?editors=1000).